client/landing/stepper/hooks/use-record-signup-complete.ts

```diff
@@ -1,37 +1,66 @@
+import {
+	isDomainRegistration,
+	isDomainTransfer,
+	isDomainMapping,
+} from '@automattic/calypso-products';
 import { useSelect } from '@wordpress/data';
 import { useCallback } from 'react';
-import { USER_STORE } from 'calypso/landing/stepper/stores';
+import { USER_STORE, ONBOARD_STORE } from 'calypso/landing/stepper/stores';
 import { recordSignupComplete } from 'calypso/lib/analytics/signup';
 import { useSite } from './use-site';
-import type { UserSelect } from '@automattic/data-stores';
+import type { UserSelect, OnboardSelect } from '@automattic/data-stores';
 
 export const useRecordSignupComplete = ( flow: string | null ) => {
 	const site = useSite();
 	const siteId = site?.ID || null;
 	const theme = site?.options?.theme_slug || '';
-	const siteCount = useSelect(
-		( select ) => ( select( USER_STORE ) as UserSelect ).getCurrentUser(),
-		[]
-	)?.site_count;
+	const { domainCartItem, planCartItem, siteCount } = useSelect( ( select ) => {
+		return {
+			siteCount: ( select( USER_STORE ) as UserSelect ).getCurrentUser()?.site_count,
+			domainCartItem: ( select( ONBOARD_STORE ) as OnboardSelect ).getDomainCartItem(),
+			planCartItem: ( select( ONBOARD_STORE ) as OnboardSelct ).getPlanCartItem(),
+		};
+	}, [] );
 
 	return useCallback( () => {
-		recordSignupComplete(
-			{
-				flow,
-				siteId: siteId,
-				isNewUser: ! siteCount,
-				hasCartItems: false,
-				isNew7DUserSite: '',
-				theme,
-				intent: flow,
-				startingPoint: flow,
-				isBlankCanvas: theme?.includes( 'blank-canvas' ),
-				domainProductSlug: '',
-				planProductSlug: '',
-				isMapping: false,
-				isTransfer: false,
-			},
-			true
```

Comment:
This `now` parameter is removed for two reasons:

1. From what I recall, it's not recommended unless there is a very good reason. The Tracks event queue is there to guarantee the order of the Tracks event is intact. By introducing `now` to force it to fire immediately, there can be edged chances that the order being ruined.
2. [The main onboarding flow](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/main.jsx#L501) has been working this way for ages :)